rust/main/submitter/src/payload_dispatcher/building_stage.rs

```diff
@@ -0,0 +1,134 @@
+// TODO: re-enable clippy warnings
+#![allow(dead_code)]
+
+use std::{collections::VecDeque, sync::Arc};
+
+use derive_new::new;
+use eyre::Result;
+use tokio::sync::{mpsc, Mutex};
+
+use crate::{payload::FullPayload, transaction::Transaction};
+
+use super::PayloadDispatcherState;
+
+pub type BuildingStageQueue = Arc<Mutex<VecDeque<FullPayload>>>;
+
+#[derive(new)]
+struct BuildingStage {
+    /// This queue is the entrypoint and event driver of the Building Stage
+    queue: BuildingStageQueue,
+    /// This channel is the exitpoint of the Building Stage
+    inclusion_stage_sender: mpsc::Sender<Transaction>,
+    state: PayloadDispatcherState,
+}
+
+impl BuildingStage {
+    pub async fn run(&self) -> Result<()> {
+        loop {
+            let payload = self.queue.lock().await.pop_front();
+            if let Some(payload) = payload {
+                let txs = self.state.adapter.build_transactions(vec![payload]).await?;
```

Comment:
I wonder if we are going to pop several payloads at once in the future so that we can benefit from batching?